src/libopenrave/kinbody.cpp

```diff
@@ -1642,128 +1642,145 @@ void KinBody::SetDOFValues(const std::vector<dReal>& vJointValues, uint32_t chec
         pJointValues = &_vTempJoints[0];
     }
 
-    boost::array<dReal,3> dummyvalues; // dummy values for a joint
-    std::vector<dReal> vtempvalues, veval;
-
-    // have to compute the angles ahead of time since they are dependent on the link transformations
-    std::vector< std::vector<dReal> > vPassiveJointValues(_vPassiveJoints.size());
-    for(size_t i = 0; i < vPassiveJointValues.size(); ++i) {
-        if( !_vPassiveJoints[i]->IsMimic() ) {
-            _vPassiveJoints[i]->GetValues(vPassiveJointValues[i]);
+    // have to compute the angles ahead of time since they are dependent on the link 
+    const int nActiveJoints = _vecjoints.size();
+    const int nPassiveJoints = _vPassiveJoints.size();
+    std::vector< std::vector<dReal> > vPassiveJointValues(nPassiveJoints);
+    for(int i = 0; i < nPassiveJoints; ++i) {
+        const KinBody::JointPtr& pjoint = _vPassiveJoints[i];
+        if(pjoint->IsStatic()) {
+            continue;
+        }
+        const boost::array<dReal, 3>& vlowerlimit = pjoint->_info._vlowerlimit;
+        const boost::array<dReal, 3>& vupperlimit = pjoint->_info._vupperlimit;
+        std::vector<dReal>& jvals = vPassiveJointValues[i];
```

Comment:
@tgn3000 although minor and i like the reference, using `vector< vector >` is slow, and i think we can speed this up even further by using `std::vector< boost::array<dReal, 3> >` instead and creating a `Joint::GetValues(boost::array<dReal,3>& values)` implementation (we can do this since the max DOF per joint is 3)